src/main.c

```diff
@@ -998,6 +998,16 @@ static gchar *info_formatter_readable(RaucManifest *manifest)
 	return g_string_free(text, FALSE);
 }
 
+static void strv_to_json_array(JsonBuilder *builder, GStrv strv)
```

Comment:
Adding input guards for `builder` would also implicitly document that `strv` can be `NULL`.

```
g_return_if_fail(JSON_IS_BUILDER(builder));
```

(rest looks fine)

Reply:
OK, I've also added the missing `#if` guard for JSON support.